saml.php

```diff
@@ -0,0 +1,95 @@
+<?php
+/* Copyright (c) 1998-2017 ILIAS open source, Extended GPL, see docs/LICENSE */
+
+/**
+ * @author Michael Jansen <mjansen@databay.de>
+ */
+
+require_once 'libs/composer/vendor/autoload.php';
+$as = new SimpleSAML_Auth_Simple('default-sp');
```

Comment:
You are right, this could be made a dynamic parameter. But reading from the database req. an initialized ILIAS context (client, database, etc.).

So moving this to let's say ilStartupGUI::doSamlAuthentication and only executing ...

<?php
require_once './Services/Context/classes/class.ilContext.php';
ilContext::init(ilContext::CONTEXT_SAML);

require_once 'Services/Init/classes/class.ilInitialisation.php';
ilInitialisation::initILIAS();

$ilCtrl->initBaseClass('ilStartUpGUI');
$ilCtrl->setCmd('doSamlAuthentication');
$ilCtrl->setTargetScript('ilias.php');
$ilCtrl->callBaseClass();
?>

... in file "saml.php" leads to a kind of chicken-and-egg problem especially if SimpleSAMLphp is configured to use the defaul "phpsession" storage (not recommended, but possible). This is also the only issue why I did not move most of the code from file "saml.php" to \ilStartUpGUI::doSamlAuthentication, yet.